README.md

```diff
@@ -167,6 +167,37 @@ steps:
       - docker#v3.3.0
 ```
 
+### Specifying secrets
+
+[Build-time variables] can be extracted from a pulled image, so when passing
+sensitive data, [secrets] should be used instead.
+
+[secrets]: https://docs.docker.com/develop/develop-images/build_enhancements/#new-docker-build-secret-information
+
+To use environment variables (perhaps fetched by another plugin) as secrets:
+
+```dockerfile
+# syntax=docker/dockerfile:1.2
+
+FROM bash
+
+RUN --mount=type=secret,id=SECRET cat /run/secrets/SECRET
+```
+
+```yaml
+steps:
+  - command: echo amaze
+    env:
+      SECRET: wow
+    plugins:
+      - seek-oss/docker-ecr-cache#v1.9.0:
+          secrets:
+            - SECRET
```

Comment:
I'd like us to land on a configuration approach that supports both `env=` and also `src=`. I think the latter will become the preferred option for npm auth when paired with https://github.com/seek-oss/private-npm-buildkite-plugin.

Reply:
How do you imagine `src=` secrets will be passed between plugins? I've decided not to implement `SECRET=value` for simplicity and because it might be visible in the pipeline. Separate PR?

Reply:
I may be misunderstanding, but in the case of `private-npm-buildkite-plugin` you should be able to control the `file` output path from that plugin and pipe that through as the input path `src=` here.

We don't have to land `src=` support in this PR, but I want to avoid ending up with strange options like `secrets` for env and `secrets-src` for src. What do you think about passing the flag values through verbatim:

```yaml
secrets:
  - id=1,env=SECRET
  - id=2,src=path/to/secret.txt
```

Alternatively we could try to retain the cleverness that your PR has currently, but I'm not sure it's worth the extra parsing complexity:


```yaml
secrets:
  - env: SECRET
  - src: path/to/secret.txt

secrets:
  id1:
    env: SECRET
  id2:
    src: path/to/secret.txt  
```

Reply:
`=` cannot occur in environment variable names, so this is forward compatible:

```yaml
secrets:
  - FOO
  - id=bar,env=BAR
  - id=baz,src=path/to/baz.txt
```